CMakeLists.txt

```diff
@@ -30,7 +30,7 @@ if(NOT CMAKE_C_STANDARD)
 endif()
 
 if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 11)
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
would this cause problems like https://github.com/getsentry/sentry-native/pull/463?

Reply:
I don’t think so, c++17 std is supported in CMake 3.10. My concern is rather the NDK version that needs an appropriate compiler.

This is just to get started, we probably want to only do this on android, since we better control the compiler versions that we want to support there.